src/Symfony/Component/ExpressionLanguage/ParserCache/ParserCacheAdapter.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\ExpressionLanguage\ParserCache;
+
+use Psr\Cache\CacheItemInterface;
+use Psr\Cache\CacheItemPoolInterface;
+use Symfony\Component\Cache\CacheItem;
+
+/**
+ * @author Alexandre GESLIN <alexandre@gesl.in>
+ *
+ * @internal
+ */
+class ParserCacheAdapter implements CacheItemPoolInterface
+{
+    private $pool;
+    private $createCacheItem;
+
+    public function __construct(ParserCacheInterface $pool)
+    {
+        $this->pool = $pool;
+
+        $this->createCacheItem = \Closure::bind(
+            function ($key, $value, $isHit) {
+                $item = new CacheItem();
+                $item->key = $key;
+                $item->value = $value;
+                $item->isHit = $isHit;
+
+                return $item;
+            },
+            null,
+            CacheItem::class
+        );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getItem($key)
+    {
+        $value = $this->pool->fetch($key);
+        $f = $this->createCacheItem;
```

Comment:
Yes, not working with `$this->createCacheItem($key, $value, $isHit);`
 picked up from here https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Cache/Adapter/ArrayAdapter.php#L75